lib/client.js

```diff
@@ -64,55 +82,71 @@ var client = function(config) {
 };
 
 /**
- * @method get
- * @param {String} ip a Joi strip.ip IP address to fetch reputation for
+ * @method getTyped
+ * @param {String} type Object type to get reputation for (ip or email)
+ * @param {String} obj Object to get reputation for
  * @return {Promise}
  */
-client.prototype.get = function (ip) {
-  if (Joi.validate(ip, ipSchema).error !== null) {
-    return Promise.reject(new Error('Invalid IP.'));
+client.prototype.getTyped = function (type, obj) {
+  if (validate(type, obj) !== null) {
+    return Promise.reject(validate(type, obj));
   }
   return this.baseRequest({
     method: 'GET',
-    uri: '/' + ip,
+    uri: '/type/' + type + '/' + obj,
     hawk: {
       contentType: null
     }
-  }).timeout(this.timeout);
+  }).timeout(this.timeout).then(function (response) {
+    if (response && response.body && response.statusCode) {
+      response.body.ip = obj;
```

Comment:
I think this line answers the question I was going to ask about compatibility but I had one other one related to this now.

So I'm assuming here we are adding the `ip` field so FxA can still handle the responses from this module without needing to be changed, as it will be expecting the old format. Should this be changed however so we only add this extra field if `type` is `ip`?

Reply:
Yes absolutely, so that we can remove the compat code in iprepd. And yeah, that makes sense to me.